d3a_interface/sim_results/energy_trade_profile.py

```diff
@@ -145,3 +151,44 @@ def convert_timestamp_strings_to_datetimes(profile):
                             outdict[k][sold_bought][dev][target][datetime_obj] = \
                                 profile[k][sold_bought][dev][target][timestamp_str]
         return outdict
+
+    @staticmethod
+    def merge_results_to_global(market_trade: Dict, global_trade: Dict, slot_list: List):
+        if not global_trade:
+            global_trade = market_trade
+            return global_trade
+        for area_uuid in market_trade:
+            if area_uuid not in global_trade or global_trade[area_uuid] == {}:
+                global_trade[area_uuid] = {"sold_energy": {}, "bought_energy": {}}
+            for sold_bought in market_trade[area_uuid]:
+                for target_area in market_trade[area_uuid][sold_bought]:
+                    if target_area not in global_trade[area_uuid][sold_bought]:
+                        global_trade[area_uuid][sold_bought][target_area] = {}
+                    for source_area in market_trade[area_uuid][sold_bought][target_area]:
+                        if source_area not in global_trade[area_uuid][sold_bought][target_area]:
+                            global_trade[area_uuid][sold_bought][target_area][source_area] = \
+                                {i: 0 for i in slot_list}
+                        global_trade[area_uuid][sold_bought][target_area][source_area].update(
+                            market_trade[area_uuid][sold_bought][target_area][source_area]
+                        )
+        return global_trade
+
+    def restore_area_results_state(self, area_uuid, last_known_state_data):
+        pass
+
+    @property
+    def plot_results(self):
+        return self.convert_timestamp_strings_to_datetimes(
+            convert_pendulum_to_str_in_dict(
+                self.traded_energy_profile, {}, ui_format=True)
+        )
```

Comment:
Sorry, I know this has been in the d3a for a while, but why do we first have to convert first to string and then to datetimes? Isn't the `plot_energy_profile` able to handle pendulum objects?

Reply:
Good question! I would also love to remove this, let me dig a bit in d3a export.py and I'll let you know what I find

Reply:
Thanks for the suggestion, I removed it and also added the call of the `add_sold_bought_lists` to this method too, in order to avoid the direct import of the class in the d3a code.